app/recipe/views.py

```diff
@@ -0,0 +1,22 @@
+from rest_framework import (
+    viewsets,
+    status,
+)
+from rest_framework.response import Response
+
+from core.recipe_models import Recipe
+from recipe import serializers
+
+
+class RecipeViewSet(viewsets.ModelViewSet):
+    serializer_class = serializers.RecipeDetailSerializer
+    queryset = Recipe.objects.all()
+
+    def get_queryset(self):
```

Comment:
There are a lot of automatic things going on with django and DRF (django rest framework) but for the name filter you'll need to do something like this: https://www.django-rest-framework.org/api-guide/filtering/#filtering-against-query-parameters